src/actions/breakpoints/syncBreakpoint.js

```diff
@@ -22,6 +24,23 @@ export async function syncClientBreakpoint(
     ? originalToGeneratedId(sourceId)
     : sourceId;
 
+  const astLocation = pendingBreakpoint.astLocation;
+  const scope = await findScopeByName(source, astLocation.name);
+
+  const scopeLocation = {
+    line: scope.location.start.line + astLocation.offset.line,
+    column: astLocation.offset.column,
+    sourceUrl: source.url,
+    sourceId
+  };
+
+  const scopeGeneratedLocation = await getGeneratedLocation(
+    getState(),
+    source,
+    scopeLocation,
+    sourceMaps
+  );
```

Comment:
this might be a bit more complicated, but should we also do the `newGeneratedLocation` so if we don't have an `astLocation` we fall back on it? Or wait, will you tell me that astLocation is doing the same thing?...

Reply:
:P